src/contracts/GPv2UniswapRouter.sol

```diff
@@ -29,6 +29,62 @@ contract GPv2UniswapRouter is UniswapV2Library {
         factory = factory_;
     }
 
+    /// @dev Execute a GPv2 settlement with a single trade against a Uniswap
+    /// path.
+    ///
+    /// This method performs no input validation, instead it performs on-chain
+    /// computation of the swap amounts in order to encode interactions and set
+    /// clearing prices such as the trader receives positive slippage. Verifying
+    /// the signature and validity of the trade (limit prices, expiry, etc.) is
+    /// done by the settlement contract.
+    ///
+    /// @param path The Uniswap route for trading.
+    /// @param trade The GPv2 trade to execute.
+    function settleSwap(IERC20[] calldata path, GPv2Trade.Data calldata trade)
```

Comment:
I think we might also want to specify a limitAmount (minReceive for sell, maxFrom for buy orders), otherwise a sophisticated frontrunner could sandwich our call and get the same limit price as the user specified.

Reply:
Hmm interesting point. Does that mean its likely that the driver would submit a solution with different slippage than user orders?

Reply:
Yes, we are already doing that today (0.1% default slippage compared to 0.5% by the user). More sophisticated slippage management is one of our key advantages over user driven trade execution even in a EIP1559 🌍 .

Reply:
Sounds good. This has a nice side effect that it also addresses your comment about early exit of the  :point_down:

Reply:
Implemented in https://github.com/gnosis/gp-v2-contracts/pull/558